### PR TITLE
Allow Ruby performance tests to use multiple server subprocesses

### PR DESCRIPTION
**Pull request by murgatroid99**

### CODE REVIEW

**Comment by murgatroid99** [comment]
test this please

File: src/ruby/qps/multiprocess_server_worker.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    end
+  end.parse!
+  p "Server worker listening on port #{port}"
+  bms = BenchmarkServer.new(config, port)
+  stopping = false
+  stop_mutex = Mutex.new
+  stop_cv = ConditionVariable.new
+  Signal.trap("USR1") do
+    stopping = true
+    stop_cv.signal
```
Comment (apolcyn) [inline src/ruby/qps/multiprocess_server_worker.rb]: Ran into this sort of situation in #9986 - I suspect this `stop_cv.signal` is causing problems?

Comment (murgatroid99) [inline src/ruby/qps/multiprocess_server_worker.rb]: Can you be more specific? I am waiting on this CV below, the condition variable `signal` method does appear to be signal safe.

Comment (apolcyn) [inline src/ruby/qps/multiprocess_server_worker.rb]: sorry scratched my comment - should be fine since it's not inside a mutex

**Comment by ctiller** [comment]
Assume this is stale

**Pull request closed without merging**